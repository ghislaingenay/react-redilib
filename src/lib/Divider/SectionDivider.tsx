import { DividerProps } from 'index';

export const SectionDivider = ({ children, testId }: DividerProps) => {
  const testIdProps = testId ? { 'data-testid': testId } : {};
  return (
    <div aria-label="Text divider" {...testIdProps}>
      {children ? (
        <div className="flex flex-row gap-x-2 items-center my-2">
          <hr className="flex flex-auto border-black border-1 m-0 box-border my-auto" />
          <h5 className="flex flex-shrink justify-center font-bold text-base uppercase">
            {children}
          </h5>
          <hr className="flex flex-auto border-black border-1 m-0 box-border my-auto" />
        </div>
      ) : (
        <hr className=" border-black border-1 m-0 my-2" />
      )}
    </div>
  );
};
