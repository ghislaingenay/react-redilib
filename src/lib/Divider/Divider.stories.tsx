import { SectionDivider } from './SectionDivider';
import type { StoryObj, Meta } from '@storybook/react';

const meta: Meta<typeof SectionDivider> = {
  title: 'Components/SectionDivider',
  component: SectionDivider,
  tags: ['autodocs'],
  argTypes: {
    children: {
      name: 'text',
      control: {
        type: 'text',
      },
    },
  },
};

export default meta;

type Story = StoryObj<typeof SectionDivider>;

export const Divider: Story = {
  render(args) {
    return <SectionDivider {...args} />;
  },
};
// export const withChildren = () => {
//   <SectionDivider>App</SectionDivider>;
// };
