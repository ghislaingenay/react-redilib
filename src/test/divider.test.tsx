import { render, screen } from '@testing-library/react';
import { SectionDivider } from '@lib/Divider';
import '@testing-library/jest-dom';

describe('Section Divider', () => {
  it('should render successfully - no children', () => {
    const { baseElement } = render(<SectionDivider />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render successfully - with children', () => {
    const { baseElement } = render(<SectionDivider>App</SectionDivider>);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render with children', () => {
    render(<SectionDivider>test</SectionDivider>);
    const getTestText = screen.getByText('test');
    expect(getTestText).toBeInTheDocument();
  });

  it('Should contains 1 hr if no children if not provided', () => {
    render(<SectionDivider />);
    const hrElement = screen.getAllByRole('separator');
    expect(hrElement).toHaveLength(1);
  });

  it('Should contains 2 hr if children provided', () => {
    render(<SectionDivider>Children</SectionDivider>);
    const hrElement = screen.getAllByRole('separator');
    expect(hrElement).toHaveLength(2);
  });

  it('can be queried by test-id with jest-dom', () => {
    render(<SectionDivider testId="test-id">Jest dom testing</SectionDivider>);
    const getTestId = screen.getByTestId('test-id');
    expect(getTestId).toBeInTheDocument();
  });

  it('show failed if the test-id is incorrect', () => {
    render(<SectionDivider testId="tesid">Jest dom testing</SectionDivider>);
    const getTestId = screen.queryByTestId('test-id');
    expect(getTestId).toBe(null);
  });

  it('can be queried by test-id with query selector', () => {
    const { container } = render(<SectionDivider testId="divider-test" />);
    const dividerElement = container.querySelector(
      'div[data-testid="divider-test"]'
    );
    expect(dividerElement).toBeInTheDocument();
  });
});
