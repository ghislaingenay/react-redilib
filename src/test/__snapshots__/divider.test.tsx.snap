// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Divider should render successfully - no children 1`] = `
<body>
  <div>
    <div
      aria-label="Text divider"
    >
      <hr
        class=" border-black border-1 m-0 my-2"
      />
    </div>
  </div>
</body>
`;

exports[`Section Divider should render successfully - with children 1`] = `
<body>
  <div>
    <div
      aria-label="Text divider"
    >
      <div
        class="flex flex-row gap-x-2 items-center my-2"
      >
        <hr
          class="flex flex-auto border-black border-1 m-0 box-border my-auto"
        />
        <h5
          class="flex flex-shrink justify-center font-bold text-base uppercase"
        >
          App
        </h5>
        <hr
          class="flex flex-auto border-black border-1 m-0 box-border my-auto"
        />
      </div>
    </div>
  </div>
</body>
`;
